slippy/tile.go

```diff
@@ -2,77 +2,123 @@ package slippy
 
 import (
 	"errors"
-	"fmt"
+	"math"
+	"strconv"
 
 	"github.com/go-spatial/geom"
 )
 
+var (
+	ErrNilBounds = errors.New("slippy: Bounds cannot be nil")
+)
+
 // MaxZoom is the lowest zoom (furthest in)
 const MaxZoom = 22
 
-// NewTile returns a Tile of Z,X,Y passed in
-func NewTile(z, x, y uint) *Tile {
-	return &Tile{
+// Zoom represents a zoom level; this usually goes from 0 to 22
+type Zoom uint
+
+func (z Zoom) N() float64 { return math.Exp2(float64(z)) }
+func (z Zoom) TileSize() Tile {
```

Comment:
What's the utility of this method and why are X and Y values both using the same value?

Reply:
It size of the number of tiles. There are n tiles in the X direction and n tiles in the Y direction.

In other srid grids the number of tiles in the Y directions can be different than in the X directions. 

We are supporting this right now but the work Julio was doing was to get support for this. He has a single structure and a ratio var, that was used to calculate the TileSize. Just moved this to each grid type here.